### ISSUE TITLE
False positive for the `new-for-builtins` rule

### ISSUE THREAD

**Issue by gigobyte** [sindresorhus/eslint-plugin-unicorn#122]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%2430%20Funded-%2300A156.svg" />](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/122)

Currently the plugin doesn't support variables that have the same name as the built-ins. 
For example:
```js
import { Map } from 'immutable'
const m = Map()
```
will report ```Use `new Map()` instead of `Map()`. (unicorn/new-for-builtins)``` which is not correct for Immutable.js

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $30.00)

- [<img src='https://avatars3.githubusercontent.com/u/44827199?v=4' alt='issuehunt' width=24 height=24> issuehunt](https://issuehunt.io/u/issuehunt) ($30.00)

### Submitted pull Requests
- [#224 Avoid false positives in new-for-builtins](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/pull/224)
- [#540 Add `isShadowed` check to `new-for-builtins`](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/pull/540)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

**Comment by sindresorhus** [comment]
It's not really a good practice to override built-ins, but I agree we should support this.

---

I think this would have been clearer:

```js
import { ImmutableMap } from 'immutable'
const m = ImmutableMap()
```

Since it's not just a normal map.

**Comment by IssuehuntBot** [comment]
@issuehuntfest has funded $30.00 to this issue. [See it on IssueHunt](https://issuehunt.io/repos/55832243/issues/122)

**Comment by alexzherdev** [comment]
@maazadeeb sorry I literally started on this an hour ago 😇

**Comment by maazadeeb** [comment]
@alexzherdev No issues.

**Comment by IssuehuntBot** [comment]
@alexzherdev has submitted a pull request. [See it on IssueHunt](https://issuehunt.io/repos/55832243/issues/122)

**Comment by sindresorhus** [comment]
This is up for grabs again. If anyone wants to tackle this, you can probably continue where https://github.com/sindresorhus/eslint-plugin-unicorn/pull/224 left off. Make sure you address the feedback given there.
